### PR TITLE
add null checks (QSH and CRYPTOCELL)

### PR DESCRIPTION
**Pull request by JacobBarthelmeh**
For GitHub issue https://github.com/wolfSSL/wolfssl/issues/2518

### CODE REVIEW

**Comment by JacobBarthelmeh** [comment]
Tesfa, can you review the CRYPTOCELL build of this and merge after done reviewing the PR.

Thanks

File: wolfcrypt/src/ecc.c [hunk 1]
```diff
@@ -4037,12 +4037,20 @@ int wc_ecc_make_key_ex(WC_RNG* rng, int keysize, ecc_key* key, int curve_id)
...
     }
 
+#if defined(WOLFSSL_CRYPTOCELL)
+    if (key->dp == NULL) {
+        WOLFSSL_MSG("ECC internal dp structure was NULL");
+        return BAD_FUNC_ARG;
+    }
+
+    raw_size = (word32) (key->dp->size)*2 + 1;
+#endif
```
Comment (tmael) [inline wolfcrypt/src/ecc.c]: Remove the block. 
Key->dp is set in the wc_ecc_reset() below it.

File: wolfcrypt/src/ecc.c [hunk 2]
```diff
@@ -7232,13 +7240,21 @@ static int wc_ecc_import_raw_private(ecc_key* key, const char* qx,
...
     }
 
+#if defined(WOLFSSL_CRYPTOCELL)
+    if (key->dp == NULL) {
+        WOLFSSL_MSG("ECC internal dp structure was NULL");
+        return BAD_FUNC_ARG;
+    }
+
+    keySz = (word32) (key->dp->size)*2 + 1;
+#endif
```
Comment (tmael) [inline wolfcrypt/src/ecc.c]: same here.

File: wolfcrypt/src/ecc.c (hunk 1 above)
Comment (JacobBarthelmeh) [reply wolfcrypt/src/ecc.c]: changed

**Comment by tmael** [comment]
Thanks @JacobBarthelmeh

**Pull request merged**